core/node/eth_watch/src/client.rs

```diff
@@ -21,10 +21,14 @@ pub trait EthClient: 'static + fmt::Debug + Send + Sync {
         &self,
         from: BlockNumber,
         to: BlockNumber,
+        topic1: H256,
+        topic2: Option<H256>,
         retries_left: usize,
     ) -> EnrichedClientResult<Vec<Log>>;
     /// Returns finalized L1 block number.
     async fn finalized_block_number(&self) -> EnrichedClientResult<u64>;
+
+    async fn get_total_priority_txs(&self) -> Result<u64, ContractCallError>;
```

Comment:
Please add comment. Also IMHO the name is confusing (I understand that it comes from contracts but still), maybe smth like `get_last_ready_priority_tx_id` will be better?

Reply:
I've changed the name to `get_number_of_processed_priority_txs`. Your name would be great, but the problem is that it's actually equal to last_ready_priority_tx - 1 as they are indexed from 0.